tokio/src/runtime/task/mod.rs

```diff
@@ -503,3 +504,14 @@ unsafe impl<S> linked_list::Link for Task<S> {
         self::core::Trailer::addr_of_owned(Header::get_trailer(target))
     }
 }
+
+/// # Safety
+///
+/// Tasks are pinned.
+unsafe impl<S> sharded_list::ShardedListItem for Task<S> {
+    unsafe fn get_shared_id(target: NonNull<Self::Target>) -> usize {
+        let task_id = unsafe { Header::get_id(target) };
+        // safety: in 32-bits machine, the lower 32 bits will be reserved
+        task_id.0 as usize
```

Comment:
I don't understand this safety comment.

Reply:
When I was writing this code, it occurred to me that `task_id` is actually different from `shard_id`. The size of `task_id` is 64 bits in all platform, but the `shard_id` depends on the specific platform: 
- 32 bits on 32-bit machine.
- 64 bits on 32-bit machine.

Casting 64 bits to 32 bits may be not safe, but we use this id for shard, so it is safe here.

Reply:
See my other comment for the right way to add these safety comments.

Reply:
make it.